### ISSUE TITLE
previewSettings width: seems like it has no effect

### ISSUE THREAD

**Issue by akluthe** [kartik-v/bootstrap-fileinput#172]
```javascript

$("#createImage_image").fileinput({
        maxFilesNum: 1,
        allowedFileExtensions: ["jpg", "pdf", "png", "gif"],
        previewSettings: {
            image: { width: "50px", height: "auto" },
            html: { width: "50px", height: "auto" },
            other: { width: "50px", height: "auto" }
        }
    });

```

In the above snippet while running version 4.1.5 in Chrome, it seems that setting the width has no effect on the width of the preview. 

I can set the height to a specific size, but the width seems to always auto.

I want a width of 50 and the height to adjust proportional to the width.

I can get sort of desirable but less specifically applied results if I change the size of the container div that the input is in. The preview sizes down accordingly that way. But it's not as specific and messes up some of my other divs. I'd rather give the preview files specific width dimensions.

Note: I am using this in a bootstrap modal.

**Comment by kartik-v** [comment]
can you isolate the problem by doing it in steps... e.g. try it without a modal  and on a full width container and then start adding constraints

**Comment by akluthe** [comment]
I certainly will. Give a me a bit and I'll come back with some more info.

**Comment by akluthe** [comment]
No matter where I put it, I can't get it the width to do anything different.

Even on a page with only the input tag on it and no container.

I notice I am getting the following inline style on the preview image.

```
style="width: 100%; height: auto;
```

I'd expect to see something other than 100% passed here if it was getting weird by being nested in some divs or a modal.

**Comment by akluthe** [comment]
How's this for weird:

I changed STYLE_SETTING to this to override everything manually.

```
STYLE_SETTING = 'style="width:50px;height:auto;"',
```

and I still get width:100% in the style.

```
<img src="blob:http%3A//localhost%3A53562/4a703104-6f37-4031-b822-e77a6c0263d5" class="file-preview-image" title="featured_peacetea_snoberry1.jpg" alt="featured_peacetea_snoberry1.jpg" style="width: 100%; height: auto;">
```

**Comment by akluthe** [comment]
If I use, Chrome Dev tools I can cahnge file-preview-frame style to get my desired results. 

It seems it was set at 97%. Searching through your code for 97% hits on this chunk of code.

```javascript

 $img.on('load', function() {
                var w1 = $thumb.width(), w2 = $preview.width();
                if (w1 > w2) {
                    $img.css('width', '100%');
                    $thumb.css('width', '97%');
                }
                var $cap = $img.closest('.file-preview-frame').find('.file-caption-name');
                if ($cap.length) {
                    $cap.width($img.width());
                    $cap.attr('title', $cap.text());
                }
                self.raise('fileimageloaded', previewId);
            });

```

setting image to width to 100% and the preview-frame to 97%.

**Comment by kartik-v** [comment]
The lines commented will work only for a specific scenario and is needed (refer past issue enhancements as requested & tested)

The condition above will fire ONLY if  the image width will NOT fit in the container for the dimensions mentioned. 

The way its designed is this way:
- The default setting is the width is auto and height is set to a value of 160px (or what you want) --- so you have thumbnails of fixed heights (since you want to fit thumbnails side by side). This will work for most images normally with 4:3 or 16:9 size resolutions.
- However, when you have very extreme dimension ratios where the width of the image is way large for the same height (for example 1000px width by 160px width) - then it will not fit the container. In this case the plugin does an additional dimension validation (as highlighted in the above code snippet you posted) --- if the width exceeds the preview container width --- then it will auto set the width of the image to be 100% so it fits within the container.

To summarize, I would suggest you to see if you could check your image dimensions - and try to set a fixed height (and optionally add a width also if needed). With height = auto you may have issues with images fitting in the container of dynamic width.
